index.js

```diff
@@ -10,9 +10,317 @@ function ask(questionText) {
 start();
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  let j = 0; //j is a switch variable. it changes based off of different responses so that the next block of code knows the implication of the previous response.
+  let min = 0; //the rest of these variables are pre-defined up here because they are used in both "game modes," to avoid using two "let"s.
+  let max = 0;
+  let tries = 0;
+  let guessMin = 0;
+  let guessMax = 0;
+
+  console.log(
+    "Salutations, honored guest! Let's play a number guessing game. One of us will think of a number, and the other will try to guess it."
+  );
+  let response = await ask(
+    'Type "my number" if you want me to guess, or type "your number" if you want to guess.\n'
+  );
+  while (true) {
```

Comment:
It would be better to run the loop based on an evaluation of a user response value, rather than setting up an infinite loop with an escape condition within it.  This will make your code more readable, and allow you to eliminate a conditional block.